text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,222 @@
+clear policy for target tiers. People not only don't know, they don't know who
+to ask or where to start.
+
+(See https://forge.rust-lang.org/platform-support.html for more information
+about targets and tiers.)
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+At a high level, the three tiers break down as follows:
+
+- Tier 3 targets provide no guarantees of support.
+- Tier 2 targets will always build, but may not pass tests.
+- Tier 1 targets will always build and pass tests.
+
+Adding a new tier 3 target imposes minimal requirements; we focus primarily on
+avoiding disruption to other ongoing Rust development.
+
+Tier 2 and tier 1 targets place work on the Rust community as a whole, to avoid
+breaking the target. Thus, these tiers require commensurate efforts from the
+maintainers of the target, to demonstrate value and to minimize any disruptions
+to ongoing Rust development.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Requirements for each tier:
+
+## Tier 3 target policy
+
+At this tier, the Rust project provides no official support for a target, so we
+place minimal requirements on the introduction of targets.
+
+- No central decision is required to add a new tier 3 target. Reviewers may
+  always use their own best judgment regarding the quality of work, and the
+  suitability of a target for the Rust project.
+- If a reviewer wishes to consult a broader team for additional guidance, they
+  should contact the compiler team.
+- If a proposed target or target-specific patch substantially changes code
+  shared with other targets (not just target-specific code), the reviewer
+  should consult the compiler team.
+- If the proposer of a target wishes to appeal the rejection of a target, they
+  may contact the compiler team.
+- Tier 3 targets must use naming consistent with any existing targets; for
+  instance, a target for a similar CPU or OS should not gratuitously use an
+  inconsistent name for that CPU or OS. Targets should normally use the same
+  names as used elsewhere in the broader ecosystem (such as in other
+  toolchains), unless they have a very good reason to diverge.
+- Tier 3 targets may have unusual requirements to build or use, but must not
+  create legal issues for the Rust project or for developers who work on those
+  targets.
+- Where possible, tier 3 targets may wish to provide documentation for the Rust
+  community for how to build and run tests for the platform, ideally using
+  emulation.
+- Tier 3 targets must not impose burden on the authors of pull requests, or
+  other developers in the community, to maintain the target. In particular,
+  do not post comments (automated or manual) on a PR that suggests a block on
+  the PR based on the target. (A PR author may choose to help with a tier 3
+  target but is not required to.)
+- Patches adding or updating tier 3 targets must not break any existing target.
```

Comment:
Is "any existing target" supposed to mean just tier 1+2 targets, or does it include other tier 3 targets too? (are there any groups of closely related tier 3 targets where this could plausibly become a real issue?)

Reply:
I would hope it includes other tier 3 targets. For example, if your target implements RV32IMC, you shouldn't be allowed to break the behaviour of atomics on a target that implements RV32GC by using target-specific features that let you emulate atomics on your platform for all RISC-V users.

Reply:
Does this mean that platform-independent changes to rustc (e.g. changing an internal `libstd` method) are allowed to break tier 3 targets, but platform-specific changes for a specific tier 3 target may not break other tier 3 targets?

Reply:
That's how I would understand it, and it seems fairly reasonable. Perhaps an explicit caveat could be added that if a tier-3 platform-specific patch would break another tier 3 target, and the author (or a reviewer, maybe?) believes it would be more beneficial to land the patch than fix the behavior on the old target, they should remove the old target as part of the patch (subject to the guidelines in the last bullet).

Reply:
I've added language discussing this.